ui/src/components/RadioGroup.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
+import cn from 'classnames';
+import CheckIcon from './icons/CheckIcon';
+
+export type RadioGroupOption = {
+  value: string;
+  label: string;
+  secondaryLabel?: string;
+  ariaLabel?: string;
+};
+
+export default function RadioGroup({
+  value,
+  setValue,
+  options,
+  defaultOption,
+}: {
+  value: string;
+  setValue: (val: string) => void;
+  options: RadioGroupOption[];
+  defaultOption?: string;
+}) {
+  return (
+    <RadioGroupPrimitive.Root
+      className="flex w-full flex-col items-start justify-start font-system-sans text-[17px]"
+      value={value}
+      defaultValue={defaultOption}
+      onValueChange={(val: string) => {
+        if (val) {
+          setValue(val);
+        }
+      }}
+    >
+      {options.map((option, index) => (
+        <RadioGroupPrimitive.Item
+          key={option.value}
+          id={option.value}
+          value={option.value}
+          aria-label={option.ariaLabel ?? option.label}
+          className={cn(
+            'flex h-14 w-full cursor-pointer items-center justify-between gap-4 border border-gray-100 py-2 px-6',
+            index === 0 ? 'rounded-t-xl' : '',
+            index === options.length - 1 ? 'rounded-b-xl' : '',
+            index !== 0 && index !== options.length - 1 ? 'border-t-0' : '',
+            index === options.length - 1 ? 'border-t-0' : '',
```

Comment:
looks like we're checking on this condition twice, can we fold that into a single check?